test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go

```diff
@@ -1,75 +1,64 @@
 	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
 	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	corev1 "k8s.io/api/core/v1"
 
 	. "github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
 	. "github.com/onsi/gomega"
-)
 
-const (
-	mustGatherPath = "../../testdata/must-gather/must-gather.e2e"
+	"github.com/jaypipes/ghw/pkg/snapshot"
 )
 
-var mustGatherContentDir string
-var snapshotDir string
-var snapshotName string
+const destDir = "must-gather"
 
 var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", func() {
-	var profile *performancev2.PerformanceProfile
+	mgContentFolder := ""
 
 	testutils.BeforeAll(func() {
-		mustgatherPathContent, err := ioutil.ReadDir(mustGatherPath)
-		Expect(err).ToNot(HaveOccurred(), "failed to read the must-gather Directory %s: %v", mustGatherPath, err)
+		destDirContent, err := ioutil.ReadDir(destDir)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "unable to read contents from destDir:%s. error: %w", destDir, err)
 
-		for _, directory := range mustgatherPathContent {
-			if strings.Contains(directory.Name(), "must-gather") {
-				mustGatherContentDir = filepath.Join(mustGatherPath, directory.Name())
-				break
+		for _, content := range destDirContent {
+			if !content.IsDir() {
+				continue
 			}
+			mgContentFolder = filepath.Join(destDir, content.Name())
 		}
-
-		// pre generated must-gather data
-		// hardcored node name
-		snapshotDir = filepath.Join(mustGatherContentDir, "nodes", "ci-ln-t0prq3t-72292-h4x2n-worker-a-wqnqt")
-		snapshotName = filepath.Join(snapshotDir, "sysinfo.tgz")
-
-		if _, err := os.Stat(snapshotName); errors.Is(err, os.ErrNotExist) {
-			Expect(err).ToNot(HaveOccurred(), "failed to read sysinfo.tgz file %s: %v", snapshotName, err)
-		}
-
-		err = Untar(snapshotDir, snapshotName)
-		Expect(err).ToNot(HaveOccurred(), "failed to read the %s: %v", snapshotName, err)
 	})
-	Context("PAO must-gather Tests", func() {
+
+	Context("with a freshly executed must-gather command", func() {
 		It("Verify Generic cluster resource definitions are captured", func() {
+
 			var genericFiles = []string{
 				"version",
 				"cluster-scoped-resources/config.openshift.io/featuregates/cluster.yaml",
 				"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/default.yaml",
 				"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/rendered.yaml",
 			}
-			err := checkfilesExist(genericFiles, mustGatherContentDir)
-			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("Checking Folder: %q\n", mgContentFolder))
+			By("\tLooking for generic files")
```

Comment:
why the initial `\t` ?

Reply:
I would like to have an answer for that. Remove it.